api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupSMCRegister.java

```diff
@@ -128,27 +141,51 @@ public String getVersion() {
         return l;
     }
 
+    private CohortDefinition getSmcUsedEncounters(){
+        CompositionCohortDefinition cd = new CompositionCohortDefinition();
+        cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+        cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+        cd.addSearch("client", ReportUtils.map(Cohorts.genderAndHasAncEncounter(false, true, "244da86d-f80e-48fe-aba9-067f241905ee"), "startDate=${startDate},endDate=${endDate}"));
+        cd.addSearch("followup", ReportUtils.map(Cohorts.genderAndHasAncEncounter(false, true, "d0f9e0b7-f336-43bd-bf50-0a7243857fa6"), "startDate=${startDate},endDate=${endDate}"));
+        cd.setCompositionString("client OR followup");
+        return cd;
+    }
+
     private DataSetDefinition dataSetDefinition() {
         PatientDataSetDefinition dsd = new PatientDataSetDefinition();
-        dsd.setName(getName());
+        dsd.setName("SMC");
         dsd.addParameters(getParameters());
-        dsd.addRowFilter(Cohorts.genderAndHasAncEncounter(false, true, "244da86d-f80e-48fe-aba9-067f241905ee"), "startDate=${startDate},endDate=${endDate}");
+        dsd.addRowFilter(getSmcUsedEncounters(), "startDate=${startDate},endDate=${endDate}");
 
         PatientIdentifierType serialNo= MetadataUtils.existing(PatientIdentifierType.class, "37601abe-2ee0-4493-8ac7-22b4972190cf");
         DataConverter identifierFormatter = new ObjectFormatter("{identifier}");
         DataDefinition identifierDef = new ConvertedPatientDataDefinition("identifier", new PatientIdentifierDataDefinition(serialNo.getName(), serialNo), identifierFormatter);
 
-        dsd.addColumn("Date", getEncounterDate(), "onOrBefore=${endDate}", new CalculationResultDataConverter());
+        dsd.addColumn("Date", getEncounterDate(), "onDate=${endDate}", new CalculationResultDataConverter());
         dsd.addColumn("Serial No", identifierDef, "");
         dsd.addColumn("Names of Client", new PreferredNameDataDefinition(), (String) null);
-        dsd.addColumn("Age<2yrs", getAgeFromEncounterDate(0, 2), "onOrBefore=${endDate}", new CalculationResultDataConverter());
-        dsd.addColumn("Age2<5yrs", getAgeFromEncounterDate(2, 5), "onOrBefore=${endDate}", new CalculationResultDataConverter());
-        dsd.addColumn("Age5<15yrs", getAgeFromEncounterDate(5, 15), "onOrBefore=${endDate}", new CalculationResultDataConverter());
-        dsd.addColumn("Age15<49yrs", getAgeFromEncounterDate(15, 49), "onOrBefore=${endDate}", new CalculationResultDataConverter());
-        dsd.addColumn("Age<49yrs", getAgeFromEncounterDate(49, 200), "onOrBefore=${endDate}", new CalculationResultDataConverter());
-        dsd.addColumn("Address", address(), "onOrBefore=${endDate}", new CalculationResultDataConverter());
+        dsd.addColumn("Age<2yrs", getAgeFromEncounterDate(0, 2), "onDate=${endDate}", new CalculationResultDataConverter());
```

Comment:
Am just reusing that method, the value of each column is dependent on that date when someone was circumcised because there is nowhere that date is recorded, that is why I did a definition that would be utilized by another definition for obtaining that date